### ISSUE TITLE
The Topology should be notified when we fail to dial a multiaddress

### ISSUE THREAD

**Issue by tomaka** [libp2p/rust-libp2p#324]
When the `PeerIdTransport` tries a multiaddress, and the dialing fails, the user should be somehow notified so that they can remove the address from the peer store.

**Comment by tomaka** [comment]
No longer relevant after #723

**Comment by tomaka** [comment]
Reopening because the intent is still there with the topology.

**Comment by tomaka** [comment]
Duplicate of #793
